tests/functional/verification/test_lock.py

```diff
@@ -0,0 +1,113 @@
+from nagini_contracts.lock import Lock
+from nagini_contracts.contracts import *
+from nagini_contracts.obligations import Level, WaitLevel
+
+
+class Cell:
+    def __init__(self, val: int) -> None:
+        self.value = val
+        Ensures(Acc(self.value) and self.value == val)
+
+
+class CellLock(Lock[Cell]):
+
+    @Predicate
+    def invariant(self) -> bool:
+        return Acc(self.get_locked().value)
+
+
+class ThingWithCell:
+    def __init__(self) -> None:
+        self.c = Cell(12)
+        self.c.value = 14
+        self.l = CellLock(self.c)
+        #:: ExpectedOutput(assignment.failed:insufficient.permission)
+        self.c.value = 16
+        Ensures(Acc(self.c) and Acc(self.l) and self.l.get_locked() is self.c)
+        Ensures(WaitLevel() < Level(self.l))
+
+    def do_a_thing(self) -> None:
+        Requires(Acc(self.l, 1/2) and Acc(self.c, 1/2) and self.l.get_locked() is self.c)
+        Requires(WaitLevel() < Level(self.l))
+        Ensures(Acc(self.l, 1 / 2) and Acc(self.c, 1 / 2))
+        #:: ExpectedOutput(postcondition.violated:assertion.false)
+        Ensures(False)
```

Comment:
The intention is to check that we cannot prove false. Since this is the postcondition, it doesn't skip anything; it just means that after the verifier has established all the postconditions that are supposed to hold, it also tries to prove false, which should never be possible. If it were able to prove this, that would indicate that there's a problem in the encoding.

Reply:
Ok, so this is a sanitizing regression test similar to what is described in the Viper tutorial, near the end. Correct? So probably a good idea to always have one such test for each new feature?